src/tcp/TcpDuplexConnection.cpp

```diff
@@ -64,7 +66,7 @@ void TcpDuplexConnection::readDataAvailable(size_t len) noexcept {
 }
 
 void TcpDuplexConnection::readEOF() noexcept {
-  inputSubscriber_.onComplete();
+  inputSubscriber_.onError(std::runtime_error("connection closed"));
```

Comment:
Is it possible to have write err and then read EOF? In this case this will SEGV, right?